### DISCUSSION TITLE
Ideas for new functionalities

### DISCUSSION THREAD

**Discussion by pointlevel** [Pythm/ad-ModeManagement#3] [Ideas]
Have a couple of ideas to share that i would like to see implemented in the ModeManagement. Just look at them as ideas.
This will be a living document.

- "turn_on_in_the_morning" 

- Support for Sonos media_players in the turn_off and turn_off lists. Doesnt support media_player.turn_off or media_player.turn_on.

**Comment by Pythm** [comment]
turn_on_in_the_morning will be added in v0.1.7 release. This uses Appdaemon turn_on service to turn on entities. I used the 'turn_off_at_night' to turn off a tv, that is a media_player entity. It checks if the entity is on before turning off, and in the morning check if entity is off before turning on. I can update the readme to better reflect that the state must be on/off.

I do not have Sonos but they are probably have a state saying inactive or media playing? This is not something that can be turned on or off, but you'll need to create a service call to start and stop.

To start a server playing media is a whole chapter in it self and will not be covered in this app.

To start media you do something like this:

```python
    self.call_service('media_player/play_media',
        entity_id = self.media_player,
        media_content_id = self.playlist,
        media_content_type = 'music'
    )

```
